app/services/tunnel.py

```
@@ -1,13 +1,32 @@
+    # Aguarda alguns instantes para que o serviço local exponha a URL.
+    time.sleep(2)
+    return proc
+
+
+def get_public_url() -> Optional[str]:
+    """Obtém a URL pública do túnel, se disponível."""
+    try:
+        resp = requests.get("http://127.0.0.1:4040/api/tunnels", timeout=5)
+        data = resp.json()
+        return data["tunnels"][0]["public_url"]
+    except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)